x/mongo/driver/topology/pool.go

```diff
@@ -358,163 +299,103 @@ func (p *pool) get(ctx context.Context) (*connection, error) {
 				Reason:  event.ReasonPoolClosed,
 			})
 		}
-		return nil, ErrPoolDisconnected
+		err = ErrPoolDisconnected
```

Comment:
Ah an interesting use-case for named returns. Seems fine to me here; I'm curious why they'd be bad. If anything, it seems like it'd be cleaner code to name returned values.

Reply:
Turns out you can use named returns _and_ return explicit values, which is easier to read I think. I've updated the code to not use "naked" return values, but the named return values are still there.

Reply:
Named return values (and the usually associated "naked" returns) can be more difficult to read because it requires the reader to keep track of the value of the named return values throughout the function. The typical explicit returns are usually easier to understand locally without having to grok the entire function.

Consider the following example:
```go
func doThing() (conn net.Conn, err error) {
	l, err := net.Listen("tcp", "localhost:0")
	// ...
	// Lots of lines of code
	// ...
	return
}
```
Will the error value returned from `net.Listen` be returned or something else assigned during the rest of the function body? You have to read the rest of the function to figure that out.

tldr; It can make local inference about return values more difficult, especially in long functions. See the official Go code review guide for [named result parameters](https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters) for more details.

Reply:
Makes sense; thanks for that in-depth explanation and the link to the official Go code review guide 🧑‍🔧